cloud-sql/mysql/servlet/src/main/java/com/example/cloudsql/TemplateData.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.cloudsql;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.List;
+import javax.sql.DataSource;
+
+public class TemplateData {
+
+  public int tabCount;
+  public int spaceCount;
+  public List<Vote> recentVotes;
+
+  public TemplateData(int tabCount, int spaceCount, List<Vote> recentVotes) {
+    this.tabCount = tabCount;
+    this.spaceCount = spaceCount;
+    this.recentVotes = recentVotes;
+  }
+
+  public static TemplateData getTemplateData(DataSource pool) throws SQLException {
+
+    int tabCount = 0;
+    int spaceCount = 0;
+    List<Vote> recentVotes = new ArrayList<>();
+    try (Connection conn = pool.getConnection()) {
```

Comment:
I'm not sure if it is worthwhile. Because isn't that just explaining how a try-with-resources works in Java? it'd be like explaining how a context manager works in Python. But if you think it'd be helpful I can add a comment.